### ISSUE TITLE
`_` redefinition check raises new errors

### ISSUE THREAD

**Issue by wch** [microsoft/pyright#3634]
In Pyright 1.1.256, we've found that many new errors are being reported for `_` in our code base, probably because of the change in 9bad0897c642d0e2bdf859cd10c09038c6416e2f.

```
 error: Function declaration "_" is obscured by a declaration of the same name (reportGeneralTypeIssues)
```

This is because we are using `_` as a placeholder name for registering callbacks. The code in question is something like this:

```py
from typing import Callable

callback_registry: list[Callable[[], None]] = []

def register(fn: Callable[[], None]) -> None:
    callback_registry.append(fn)

@register
def _():
    print("Callback 1 called")

@register
def _():
    print("Callback 2 called")
```

When run through pyright 1.1.256, it says:

```
..../test.py:11:5 - error: Function declaration "_" is obscured by a declaration of the same name (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```

In our use case, we're not actually using the name `_` -- we just had to give the function _some_ name because Python doesn't support anonymous functions with more than one line.

Is it intentional that pyright will report an error in cases like this?

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.257, which I just published. It will also be included in the next release of pylance.
